lib/shoulda/matchers/active_model/validate_confirmation_of_matcher.rb

```diff
@@ -94,7 +94,7 @@ def simple_description
 
         def matches?(subject)
           super(subject)
-
+          @data_type = @subject.class.columns_hash[@attribute.to_s].type
```

Comment:
Let's extract this to a method (called `column_type` instead) instead of making this an instance variable, I think that will ensure there's not too much clutter here.

Also, note that this assumes that the model is an ActiveRecord model, and I think we can't make that assumption, because otherwise this will fail. If you look at [other matchers](https://github.com/thoughtbot/shoulda-matchers/blob/master/lib/shoulda/matchers/active_model/validate_absence_of_matcher.rb#L117) you'll see how we do this. Obviously, this means that we will also need a test that the confirmation matcher works with an ActiveModel model. I'm not interested in comprehensive tests right now, so a simple example should work fine ([here](https://github.com/thoughtbot/shoulda-matchers/blob/master/spec/unit/shoulda/matchers/active_model/validate_absence_of_matcher_spec.rb#L67) is an example of that).